pwndbg/commands/context.py

```diff
@@ -180,23 +196,52 @@ def context_code():
         if closest_line < 0:
             return []
 
-        source = gdb.execute('list %i' % closest_line, from_tty=False, to_string=True)
+        # Get the full source code
+        filename = symtab.fullname()
+        source = get_highlight_source(filename)
 
         # If it starts on line 1, it's not really using the
         # correct source code.
         if not source or closest_line <= 1:
             return []
 
-        # highlight the current code line
-        source_lines = source.splitlines()
-        if pwndbg.config.highlight_source:
-            for i in range(len(source_lines)):
-                if source_lines[i].startswith('%s\t' % closest_line):
-                    source_lines[i] = C.highlight(source_lines[i])
-                    break
-
-        banner = [pwndbg.ui.banner("source")]
-        banner.extend(source_lines)
+        n = int(source_code_lines)
+
+        # Compute the line range
+        start = max(closest_line - 1 - n//2, 0)
+        end = min(closest_line - 1 + n//2 + 1, len(source))
+        num_width = len(str(end))
+
+        # split the code
+        source = source[start:end]
+
+        # Format the output
+        formatted_source = []
+        for line_number, code in enumerate(source, start=start + 1):
+            # Compute the prefix
+            # XXX: use nearpc-prefix here, provide customize?
+
+            # XXX: copy from color/disasm
+            prefix_sign = pwndbg.config.nearpc_prefix
+            value = prefix_sign.value
+            if isinstance(value, bytes):
+                value = codecs.decode(value, 'utf-8')
+
+            fmt = ' {prefix_sign:{prefix_width}} {line_number:>{num_width}} {code}'
+            if pwndbg.config.highlight_source and line_number == closest_line:
+                fmt = C.highlight(fmt)
+
+            line = fmt.format(
+                prefix_sign = prefix_sign if line_number == closest_line else '',
```

Comment:
This line must be:
```python
prefix_sign = str(prefix_sign) if line_number == closest_line else '',
```

Otherwise I am getting:
```
Traceback (most recent call last):
  File "pwndbg/pwndbg/commands/__init__.py", line 107, in __call__
    return self.function(*args, **kwargs)
  File "pwndbg/pwndbg/commands/__init__.py", line 198, in _OnlyWhenRunning
    return function(*a, **kw)
  File "pwndbg/pwndbg/commands/context.py", line 65, in context
    result.extend(func())
  File "pwndbg/pwndbg/commands/context.py", line 242, in context_code
    code = code
TypeError: unsupported format string passed to Parameter.__format__
```

Reply:
Also can we remove spaces before and after `=`?
Its not really variable assignment and just a keyword arguments passing and this is usually done without spaces huh

Reply:
Both Fixed.
For Parameter issue, maybe we can implement `__format__` method to support this case and then we can remove all operation of casting to `str` or `int`.